com.ericsson.otp.erlang.OtpErlangExit: badmatch in debugger

# System

Plugin Version: 13.1.0-pre+20220531183708
Application: IntelliJ IDEA Community Edition (2022.1)
Operating System: Mac OS X (12.4)


# Event


## Exception


### Stacktrace


<details>

<summary>
java.security.PrivilegedActionException: java.security.PrivilegedActionException: java.util.concurrent.ExecutionException: com.ericsson.otp.erlang.OtpErlangExit: 

```
{
  {
    badmatch,
    {
     {
      conn,
      [{version,0},{line,1}],nil},{elixir_ex,false,warn,false,{#{{conn,0} => false},2},{#{{conn,nil} => 0,{user_params,nil} => 1},false}},#{'__struct__' => 'Elixir.Macro.Env',aliases => [],context => nil,context_modules => [],file => #Bin<89>,function => {create,2},functions => [{'Elixir.Kernel',[{'!=',2},{'!==',2},{'*',2},{'**',2},{'+',1},{'+',2},{'++',2},{'-',1},{'-',2},{'--',2},{'/',2},{'<',2},{'<=',2},{'==',2},{'===',2},{'=~',2},{'>',2},{'>=',2},{abs,1},{apply,2},{apply,3},{binary_part,3},{bit_size,1},{byte_size,1},{ceil,1},{div,2},{elem,2},{exit,1},{floor,1},{'function_exported?',3},{get_and_update_in,3},{get_in,2},{hd,1},{inspect,1},{inspect,2},{is_atom,1},{is_binary,1},{is_bitstring,1},{is_boolean,1},{is_float,1},{is_function,1},{is_function,2},{is_integer,1},{is_list,1},{is_map,1},{is_map_key,2},{is_number,1},{is_pid,1},{is_port,1},{is_reference,1},{is_tuple,1},{length,1},{'macro_exported?',3},{make_ref,0},{map_size,1},{max,2},{min,2},{node,0},{node,1},{not,1},{pop_in,2},{put_elem,3},{put_in,3},{rem,2},{round,1},{self,0},{send,2},{spawn,1},{spawn,3},{spawn_link,1},{spawn_link,3},{spawn_monitor,1},{spawn_monitor,3},{struct,1},{struct,2},{'struct!',1},{'struct!',2},{throw,1},{tl,1},{trunc,1},{tuple_size,1},{update_in,3}]}],lexical_tracker => nil,line => 22,macro_aliases => [],macros => [{'Elixir.Kernel',[{'!',1},{'&&',2},{'..',2},{'..//',3},{'<>',2},{'@',1},{'alias!',1},{and,2},{binding,0},{binding,1},{def,1},{def,2},{defdelegate,2},{defexception,1},{defguard,1},{defguardp,1},{defimpl,2},{defimpl,3},{defmacro,1},{defmacro,2},{defmacrop,1},{defmacrop,2},{defmodule,2},{defoverridable,1},{defp,1},{defp,2},{defprotocol,2},{defstruct,1},{destructure,2},{get_and_update_in,2},{if,2},{in,2},{is_exception,1},{is_exception,2},{is_nil,1},{is_struct,1},{is_struct,2},{'match?',2},{or,2},{pop_in,1},{put_in,2},{raise,1},{raise,2},{reraise,2},{reraise,3},{sigil_C,2},{sigil_D,2},{sigil_N,2},{sigil_R,2},{sigil_S,2},{sigil_T,2},{sigil_U,2},{sigil_W,2},{sigil_c,2},{sigil_r,2},{sigil_s,2},{sigil_w,2},{tap,2},{then,2},{to_char_list,1},{to_charlist,1},{to_string,1},{unless,2},{update_in,2},{use,1},{use,2},{'var!',1},{'var!',2},{'|>',2},{'||',2}]}],module => 'Elixir.LevelWeb.SessionController',requires => ['Elixir.Application','Elixir.Kernel','Elixir.Kernel.Typespec'],tracers => [],versioned_vars => #{{conn,nil} => 0,{user_params,nil} => 1}}}},[{'Elixir.IntelliJElixir.Debugger.Server',quoted_to_erl,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,419}]},{'Elixir.IntelliJElixir.Debugger.Server',handle_call,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,284}]},{gen_server,try_handle_call,4,[{file,"gen_server.erl"},{line,721}]},{gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,750}]},{proc_lib,init_p_do_apply,3, [{file,"proc_lib.erl"},{line,226}]}]
}
```

</summary>

```
java.security.PrivilegedActionException: java.security.PrivilegedActionException: java.util.concurrent.ExecutionException: com.ericsson.otp.erlang.OtpErlangExit: {{badmatch,{{conn,[{version,0},{line,1}],nil},{elixir_ex,false,warn,false,{#{{conn,0} => false},2},{#{{conn,nil} => 0,{user_params,nil} => 1},false}},#{'__struct__' => 'Elixir.Macro.Env',aliases => [],context => nil,context_modules => [],file => #Bin<89>,function => {create,2},functions => [{'Elixir.Kernel',[{'!=',2},{'!==',2},{'*',2},{'**',2},{'+',1},{'+',2},{'++',2},{'-',1},{'-',2},{'--',2},{'/',2},{'<',2},{'<=',2},{'==',2},{'===',2},{'=~',2},{'>',2},{'>=',2},{abs,1},{apply,2},{apply,3},{binary_part,3},{bit_size,1},{byte_size,1},{ceil,1},{div,2},{elem,2},{exit,1},{floor,1},{'function_exported?',3},{get_and_update_in,3},{get_in,2},{hd,1},{inspect,1},{inspect,2},{is_atom,1},{is_binary,1},{is_bitstring,1},{is_boolean,1},{is_float,1},{is_function,1},{is_function,2},{is_integer,1},{is_list,1},{is_map,1},{is_map_key,2},{is_number,1},{is_pid,1},{is_port,1},{is_reference,1},{is_tuple,1},{length,1},{'macro_exported?',3},{make_ref,0},{map_size,1},{max,2},{min,2},{node,0},{node,1},{not,1},{pop_in,2},{put_elem,3},{put_in,3},{rem,2},{round,1},{self,0},{send,2},{spawn,1},{spawn,3},{spawn_link,1},{spawn_link,3},{spawn_monitor,1},{spawn_monitor,3},{struct,1},{struct,2},{'struct!',1},{'struct!',2},{throw,1},{tl,1},{trunc,1},{tuple_size,1},{update_in,3}]}],lexical_tracker => nil,line => 22,macro_aliases => [],macros => [{'Elixir.Kernel',[{'!',1},{'&&',2},{'..',2},{'..//',3},{'<>',2},{'@',1},{'alias!',1},{and,2},{binding,0},{binding,1},{def,1},{def,2},{defdelegate,2},{defexception,1},{defguard,1},{defguardp,1},{defimpl,2},{defimpl,3},{defmacro,1},{defmacro,2},{defmacrop,1},{defmacrop,2},{defmodule,2},{defoverridable,1},{defp,1},{defp,2},{defprotocol,2},{defstruct,1},{destructure,2},{get_and_update_in,2},{if,2},{in,2},{is_exception,1},{is_exception,2},{is_nil,1},{is_struct,1},{is_struct,2},{'match?',2},{or,2},{pop_in,1},{put_in,2},{raise,1},{raise,2},{reraise,2},{reraise,3},{sigil_C,2},{sigil_D,2},{sigil_N,2},{sigil_R,2},{sigil_S,2},{sigil_T,2},{sigil_U,2},{sigil_W,2},{sigil_c,2},{sigil_r,2},{sigil_s,2},{sigil_w,2},{tap,2},{then,2},{to_char_list,1},{to_charlist,1},{to_string,1},{unless,2},{update_in,2},{use,1},{use,2},{'var!',1},{'var!',2},{'|>',2},{'||',2}]}],module => 'Elixir.LevelWeb.SessionController',requires => ['Elixir.Application','Elixir.Kernel','Elixir.Kernel.Typespec'],tracers => [],versioned_vars => #{{conn,nil} => 0,{user_params,nil} => 1}}}},[{'Elixir.IntelliJElixir.Debugger.Server',quoted_to_erl,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,419}]},{'Elixir.IntelliJElixir.Debugger.Server',handle_call,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,284}]},{gen_server,try_handle_call,4,[{file,"gen_server.erl"},{line,721}]},{gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,750}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:898)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:820)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:743)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:439)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:803)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:438)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:119)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:604)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:436)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:873)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:484)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.security.PrivilegedActionException: java.util.concurrent.ExecutionException: com.ericsson.otp.erlang.OtpErlangExit: {{badmatch,{{conn,[{version,0},{line,1}],nil},{elixir_ex,false,warn,false,{#{{conn,0} => false},2},{#{{conn,nil} => 0,{user_params,nil} => 1},false}},#{'__struct__' => 'Elixir.Macro.Env',aliases => [],context => nil,context_modules => [],file => #Bin<89>,function => {create,2},functions => [{'Elixir.Kernel',[{'!=',2},{'!==',2},{'*',2},{'**',2},{'+',1},{'+',2},{'++',2},{'-',1},{'-',2},{'--',2},{'/',2},{'<',2},{'<=',2},{'==',2},{'===',2},{'=~',2},{'>',2},{'>=',2},{abs,1},{apply,2},{apply,3},{binary_part,3},{bit_size,1},{byte_size,1},{ceil,1},{div,2},{elem,2},{exit,1},{floor,1},{'function_exported?',3},{get_and_update_in,3},{get_in,2},{hd,1},{inspect,1},{inspect,2},{is_atom,1},{is_binary,1},{is_bitstring,1},{is_boolean,1},{is_float,1},{is_function,1},{is_function,2},{is_integer,1},{is_list,1},{is_map,1},{is_map_key,2},{is_number,1},{is_pid,1},{is_port,1},{is_reference,1},{is_tuple,1},{length,1},{'macro_exported?',3},{make_ref,0},{map_size,1},{max,2},{min,2},{node,0},{node,1},{not,1},{pop_in,2},{put_elem,3},{put_in,3},{rem,2},{round,1},{self,0},{send,2},{spawn,1},{spawn,3},{spawn_link,1},{spawn_link,3},{spawn_monitor,1},{spawn_monitor,3},{struct,1},{struct,2},{'struct!',1},{'struct!',2},{throw,1},{tl,1},{trunc,1},{tuple_size,1},{update_in,3}]}],lexical_tracker => nil,line => 22,macro_aliases => [],macros => [{'Elixir.Kernel',[{'!',1},{'&&',2},{'..',2},{'..//',3},{'<>',2},{'@',1},{'alias!',1},{and,2},{binding,0},{binding,1},{def,1},{def,2},{defdelegate,2},{defexception,1},{defguard,1},{defguardp,1},{defimpl,2},{defimpl,3},{defmacro,1},{defmacro,2},{defmacrop,1},{defmacrop,2},{defmodule,2},{defoverridable,1},{defp,1},{defp,2},{defprotocol,2},{defstruct,1},{destructure,2},{get_and_update_in,2},{if,2},{in,2},{is_exception,1},{is_exception,2},{is_nil,1},{is_struct,1},{is_struct,2},{'match?',2},{or,2},{pop_in,1},{put_in,2},{raise,1},{raise,2},{reraise,2},{reraise,3},{sigil_C,2},{sigil_D,2},{sigil_N,2},{sigil_R,2},{sigil_S,2},{sigil_T,2},{sigil_U,2},{sigil_W,2},{sigil_c,2},{sigil_r,2},{sigil_s,2},{sigil_w,2},{tap,2},{then,2},{to_char_list,1},{to_charlist,1},{to_string,1},{unless,2},{update_in,2},{use,1},{use,2},{'var!',1},{'var!',2},{'|>',2},{'||',2}]}],module => 'Elixir.LevelWeb.SessionController',requires => ['Elixir.Application','Elixir.Kernel','Elixir.Kernel.Typespec'],tracers => [],versioned_vars => #{{conn,nil} => 0,{user_params,nil} => 1}}}},[{'Elixir.IntelliJElixir.Debugger.Server',quoted_to_erl,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,419}]},{'Elixir.IntelliJElixir.Debugger.Server',handle_call,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,284}]},{gen_server,try_handle_call,4,[{file,"gen_server.erl"},{line,721}]},{gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,750}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	... 20 more
Caused by: java.util.concurrent.ExecutionException: com.ericsson.otp.erlang.OtpErlangExit: {{badmatch,{{conn,[{version,0},{line,1}],nil},{elixir_ex,false,warn,false,{#{{conn,0} => false},2},{#{{conn,nil} => 0,{user_params,nil} => 1},false}},#{'__struct__' => 'Elixir.Macro.Env',aliases => [],context => nil,context_modules => [],file => #Bin<89>,function => {create,2},functions => [{'Elixir.Kernel',[{'!=',2},{'!==',2},{'*',2},{'**',2},{'+',1},{'+',2},{'++',2},{'-',1},{'-',2},{'--',2},{'/',2},{'<',2},{'<=',2},{'==',2},{'===',2},{'=~',2},{'>',2},{'>=',2},{abs,1},{apply,2},{apply,3},{binary_part,3},{bit_size,1},{byte_size,1},{ceil,1},{div,2},{elem,2},{exit,1},{floor,1},{'function_exported?',3},{get_and_update_in,3},{get_in,2},{hd,1},{inspect,1},{inspect,2},{is_atom,1},{is_binary,1},{is_bitstring,1},{is_boolean,1},{is_float,1},{is_function,1},{is_function,2},{is_integer,1},{is_list,1},{is_map,1},{is_map_key,2},{is_number,1},{is_pid,1},{is_port,1},{is_reference,1},{is_tuple,1},{length,1},{'macro_exported?',3},{make_ref,0},{map_size,1},{max,2},{min,2},{node,0},{node,1},{not,1},{pop_in,2},{put_elem,3},{put_in,3},{rem,2},{round,1},{self,0},{send,2},{spawn,1},{spawn,3},{spawn_link,1},{spawn_link,3},{spawn_monitor,1},{spawn_monitor,3},{struct,1},{struct,2},{'struct!',1},{'struct!',2},{throw,1},{tl,1},{trunc,1},{tuple_size,1},{update_in,3}]}],lexical_tracker => nil,line => 22,macro_aliases => [],macros => [{'Elixir.Kernel',[{'!',1},{'&&',2},{'..',2},{'..//',3},{'<>',2},{'@',1},{'alias!',1},{and,2},{binding,0},{binding,1},{def,1},{def,2},{defdelegate,2},{defexception,1},{defguard,1},{defguardp,1},{defimpl,2},{defimpl,3},{defmacro,1},{defmacro,2},{defmacrop,1},{defmacrop,2},{defmodule,2},{defoverridable,1},{defp,1},{defp,2},{defprotocol,2},{defstruct,1},{destructure,2},{get_and_update_in,2},{if,2},{in,2},{is_exception,1},{is_exception,2},{is_nil,1},{is_struct,1},{is_struct,2},{'match?',2},{or,2},{pop_in,1},{put_in,2},{raise,1},{raise,2},{reraise,2},{reraise,3},{sigil_C,2},{sigil_D,2},{sigil_N,2},{sigil_R,2},{sigil_S,2},{sigil_T,2},{sigil_U,2},{sigil_W,2},{sigil_c,2},{sigil_r,2},{sigil_s,2},{sigil_w,2},{tap,2},{then,2},{to_char_list,1},{to_charlist,1},{to_string,1},{unless,2},{update_in,2},{use,1},{use,2},{'var!',1},{'var!',2},{'|>',2},{'||',2}]}],module => 'Elixir.LevelWeb.SessionController',requires => ['Elixir.Application','Elixir.Kernel','Elixir.Kernel.Typespec'],tracers => [],versioned_vars => #{{conn,nil} => 0,{user_params,nil} => 1}}}},[{'Elixir.IntelliJElixir.Debugger.Server',quoted_to_erl,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,419}]},{'Elixir.IntelliJElixir.Debugger.Server',handle_call,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,284}]},{gen_server,try_handle_call,4,[{file,"gen_server.erl"},{line,721}]},{gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,750}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2022)
	at org.elixir_lang.MailBox.wait(MailBox.kt:203)
	at org.elixir_lang.MailBox.receive(MailBox.kt:177)
	at org.elixir_lang.MailBox.genericUnmonitoredCall(MailBox.kt:127)
	at org.elixir_lang.MailBox.genericUnmonitoredCall(MailBox.kt:114)
	at org.elixir_lang.MailBox.genericServerUnmonitoredCall(MailBox.kt:101)
	at org.elixir_lang.MailBox.genericServerCall(MailBox.kt:45)
	at org.elixir_lang.MailBox.genericServerCall(MailBox.kt:39)
	at org.elixir_lang.debugger.Node.callDebugged(Node.kt:255)
	at org.elixir_lang.debugger.Node.evaluate(Node.kt:206)
	at org.elixir_lang.debugger.Process.evaluate(Process.kt:431)
	at org.elixir_lang.debugger.Evaluator.evaluate(Evaluator.kt:19)
	at com.intellij.xdebugger.evaluation.XDebuggerEvaluator.evaluate(XDebuggerEvaluator.java:35)
	at com.intellij.xdebugger.impl.evaluate.XDebuggerEvaluationDialog.startEvaluation(XDebuggerEvaluationDialog.java:344)
	at com.intellij.xdebugger.impl.ui.tree.nodes.EvaluatingExpressionRootNode$EvaluatingResultContainer.computeChildren(EvaluatingExpressionRootNode.java:33)
	at com.intellij.xdebugger.impl.ui.tree.nodes.XValueContainerNode.startComputingChildren(XValueContainerNode.java:57)
	at com.intellij.xdebugger.impl.ui.tree.nodes.XValueContainerNode.loadChildren(XValueContainerNode.java:47)
	at com.intellij.xdebugger.impl.ui.tree.nodes.XValueContainerNode.getChildren(XValueContainerNode.java:263)
	at com.intellij.xdebugger.impl.ui.tree.nodes.XDebuggerTreeNode.getChildCount(XDebuggerTreeNode.java:52)
	at java.desktop/javax.swing.tree.DefaultTreeModel.getChildCount(DefaultTreeModel.java:192)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1457)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1272)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.rebuild(VariableHeightLayoutCache.java:728)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.treeStructureChanged(VariableHeightLayoutCache.java:629)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$Handler.treeStructureChanged(BasicTreeUI.java:4324)
	at java.desktop/javax.swing.tree.DefaultTreeModel.fireTreeStructureChanged(DefaultTreeModel.java:607)
	at java.desktop/javax.swing.tree.DefaultTreeModel.nodeStructureChanged(DefaultTreeModel.java:391)
	at java.desktop/javax.swing.tree.DefaultTreeModel.setRoot(DefaultTreeModel.java:141)
	at com.intellij.xdebugger.impl.ui.tree.XDebuggerTree.setRoot(XDebuggerTree.java:298)
	at com.intellij.xdebugger.impl.evaluate.XDebuggerEvaluationDialog.evaluate(XDebuggerEvaluationDialog.java:297)
	at com.intellij.xdebugger.impl.evaluate.XDebuggerEvaluationDialog.doOKAction(XDebuggerEvaluationDialog.java:189)
	at com.intellij.openapi.ui.DialogWrapper$OkAction.doAction(DialogWrapper.java:1853)
	at com.intellij.openapi.ui.DialogWrapper$DialogWrapperAction.actionPerformed(DialogWrapper.java:1807)
	at com.intellij.xdebugger.impl.evaluate.XDebuggerEvaluationDialog$4.actionPerformed(XDebuggerEvaluationDialog.java:198)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:270)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6654)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345)
	at java.desktop/java.awt.Component.processEvent(Component.java:6419)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5029)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4861)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2793)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4861)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	... 25 more
Caused by: com.ericsson.otp.erlang.OtpErlangExit: {{badmatch,{{conn,[{version,0},{line,1}],nil},{elixir_ex,false,warn,false,{#{{conn,0} => false},2},{#{{conn,nil} => 0,{user_params,nil} => 1},false}},#{'__struct__' => 'Elixir.Macro.Env',aliases => [],context => nil,context_modules => [],file => #Bin<89>,function => {create,2},functions => [{'Elixir.Kernel',[{'!=',2},{'!==',2},{'*',2},{'**',2},{'+',1},{'+',2},{'++',2},{'-',1},{'-',2},{'--',2},{'/',2},{'<',2},{'<=',2},{'==',2},{'===',2},{'=~',2},{'>',2},{'>=',2},{abs,1},{apply,2},{apply,3},{binary_part,3},{bit_size,1},{byte_size,1},{ceil,1},{div,2},{elem,2},{exit,1},{floor,1},{'function_exported?',3},{get_and_update_in,3},{get_in,2},{hd,1},{inspect,1},{inspect,2},{is_atom,1},{is_binary,1},{is_bitstring,1},{is_boolean,1},{is_float,1},{is_function,1},{is_function,2},{is_integer,1},{is_list,1},{is_map,1},{is_map_key,2},{is_number,1},{is_pid,1},{is_port,1},{is_reference,1},{is_tuple,1},{length,1},{'macro_exported?',3},{make_ref,0},{map_size,1},{max,2},{min,2},{node,0},{node,1},{not,1},{pop_in,2},{put_elem,3},{put_in,3},{rem,2},{round,1},{self,0},{send,2},{spawn,1},{spawn,3},{spawn_link,1},{spawn_link,3},{spawn_monitor,1},{spawn_monitor,3},{struct,1},{struct,2},{'struct!',1},{'struct!',2},{throw,1},{tl,1},{trunc,1},{tuple_size,1},{update_in,3}]}],lexical_tracker => nil,line => 22,macro_aliases => [],macros => [{'Elixir.Kernel',[{'!',1},{'&&',2},{'..',2},{'..//',3},{'<>',2},{'@',1},{'alias!',1},{and,2},{binding,0},{binding,1},{def,1},{def,2},{defdelegate,2},{defexception,1},{defguard,1},{defguardp,1},{defimpl,2},{defimpl,3},{defmacro,1},{defmacro,2},{defmacrop,1},{defmacrop,2},{defmodule,2},{defoverridable,1},{defp,1},{defp,2},{defprotocol,2},{defstruct,1},{destructure,2},{get_and_update_in,2},{if,2},{in,2},{is_exception,1},{is_exception,2},{is_nil,1},{is_struct,1},{is_struct,2},{'match?',2},{or,2},{pop_in,1},{put_in,2},{raise,1},{raise,2},{reraise,2},{reraise,3},{sigil_C,2},{sigil_D,2},{sigil_N,2},{sigil_R,2},{sigil_S,2},{sigil_T,2},{sigil_U,2},{sigil_W,2},{sigil_c,2},{sigil_r,2},{sigil_s,2},{sigil_w,2},{tap,2},{then,2},{to_char_list,1},{to_charlist,1},{to_string,1},{unless,2},{update_in,2},{use,1},{use,2},{'var!',1},{'var!',2},{'|>',2},{'||',2}]}],module => 'Elixir.LevelWeb.SessionController',requires => ['Elixir.Application','Elixir.Kernel','Elixir.Kernel.Typespec'],tracers => [],versioned_vars => #{{conn,nil} => 0,{user_params,nil} => 1}}}},[{'Elixir.IntelliJElixir.Debugger.Server',quoted_to_erl,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,419}]},{'Elixir.IntelliJElixir.Debugger.Server',handle_call,3,[{file,"/private/var/folders/71/j16_4v856_5dsfl3bcm2v5lr0000gp/T/intellij_elixir10/debugger/lib/intellij_elixir/debugger/server.ex"},{line,284}]},{gen_server,try_handle_call,4,[{file,"gen_server.erl"},{line,721}]},{gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,750}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}
	at com.ericsson.otp.erlang.OtpMbox.receiveMsg(OtpMbox.java:272)
	at com.ericsson.otp.erlang.OtpMbox.receive(OtpMbox.java:168)
	at org.elixir_lang.MailBox.loop(MailBox.kt:219)
	at org.elixir_lang.MailBox._init_$lambda-0(MailBox.kt:26)
```

</details>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -296,6 +296,8 @@
       3. Click the `mix format` tab
       4. Expand the General group
       5. Uncheck "Format files with `mix format`".
+* [#2697](https://github.com/KronicDeth/intellij-elixir/pull/2697) - [@KronicDeth](https://github.com/KronicDeth)
+  * Support Elixir 1.13.0 in debugger.
 
 ### Bug Fixes
 
@@ -363,6 +365,9 @@
 * [#2695](https://github.com/KronicDeth/intellij-elixir/pull/2695) - [@KronicDeth](https://github.com/KronicDeth)
   * Restrict `UsageTargetProvider` to `ElixirFile`s
     Without this restriction, it tries to run when developing the plugin itself and breaks Kotlin syntax highlighting.
+* [#2697](https://github.com/KronicDeth/intellij-elixir/pull/2697) - [@KronicDeth](https://github.com/KronicDeth)
+  * Fix environment not being passed to debug runs of ESpec and ExUnit Run Configurations.
+    The `env` from the `Configuration` was dropped because a local `env` was created to set `MIX_ENV` `true`.
 
 ## v13.0.0
 

--- a/resources/META-INF/changelog.html
+++ b/resources/META-INF/changelog.html
@@ -52,6 +52,7 @@
           </li>
         </ul>
       </li>
+      <li>Support Elixir 1.13.0 in debugger.</li>
     </ul>
   </li>
   <li>
@@ -126,6 +127,8 @@
       <li>Remove <code class="notranslate">CodeStyleSettingsProvider</code> because it is redundant with <code class="notranslate">LanguageCodeStyleSettingsProvider</code></li>
       <li>Restrict <code class="notranslate">UsageTargetProvider</code> to <code class="notranslate">ElixirFile</code>s<br>
         Without this restriction, it tries to run when developing the plugin itself and breaks Kotlin syntax highlighting.</li>
+      <li>Fix environment not being passed to debug runs of ESpec and ExUnit Run Configurations.<br>
+        The <code class="notranslate">env</code> from the <code class="notranslate">Configuration</code> was dropped because a local <code class="notranslate">env</code> was created to set <code class="notranslate">MIX_ENV</code> <code class="notranslate">true</code>.</li>
     </ul>
   </li>
 </ul>
@@ -893,23 +896,6 @@
         Walk map constructin arguments, associatons, and variables when resolving type parameters.
       </li>
       <li>Don't use <code>PluginId.findId</code> that doesn't exist in 2021.1.X</li>
-    </ul>
-  </li>
-</ul>
-<h1>v11.13.0</h1>
-<ul>
-  <li>
-    <p>Bug Fixes</p>
-    <ul>
-      <li>Remove bintray repositories.</li>
-    </ul>
-  </li>
-  <li>
-    <p>Enhancements</p>
-    <ul>
-      <li>Update build to IntelliJ IDEA 2021.2.</li>
-      <li>Upgrade to Gradle 7.1.1.</li>
-      <li>Expand compatibility range to 2021.1.2-2021.2.</li>
     </ul>
   </li>
 </ul>

--- a/resources/debugger/lib/intellij_elixir/debugger/server.ex
+++ b/resources/debugger/lib/intellij_elixir/debugger/server.ex
@@ -37,7 +37,7 @@ defmodule IntelliJElixir.Debugger.Server do
         stacktrace: false
       )
 
-    true ->
+    Version.compare(System.version(), "1.10.0") == :lt ->
       # https://github.com/elixir-lang/elixir/blob/v1.9.2/lib/elixir/src/elixir.hrl#L8-L18
       Record.defrecordp(
         :elixir_erl,
@@ -50,6 +50,19 @@ defmodule IntelliJElixir.Debugger.Server do
         counter: %{},
         expand_captures: false,
         stacktrace: false
+      )
+    true ->
+      # https://github.com/elixir-lang/elixir/blob/v1.10.0/lib/elixir/src/elixir.hrl#L8-L17
+      Record.defrecordp(
+        :elixir_erl,
+        context: nil,
+        extra: nil,
+        caller: false,
+        var_names: %{},
+        extra_guards: [],
+        counter: %{},
+        expand_captures: false,
+        stacktrace: nil
       )
   end
 
@@ -256,16 +269,8 @@ defmodule IntelliJElixir.Debugger.Server do
                 {elixir_variable_name, nil}
               end)
 
-            parsed_scope =
-              elixir_erl(
-                vars: vars(elixir_variable_tuples),
-                counter: counter(elixir_variable_tuples)
-              )
-
-            # https://github.com/elixir-lang/elixir/blob/8a971fcb44391bd8b16456666f3033b633c6ff77/lib/elixir/src/elixir.erl#L255
-            # Elixir 1.7+ has :elixir_env.with_vars, but can't use here for Elixir 1.6.5 compatibility, so use
-            # https://github.com/elixir-lang/elixir/blob/v1.6.5/lib/elixir/src/elixir.erl#L223
-            vars_env = %{env | vars: parsed_vars}
+            parsed_scope = parsed_scope(elixir_variable_tuples)
+            vars_env = elixir_env_with_vars(env, parsed_vars)
 
             # Elixir 1.7+ uses current_vars
             current_vars_env =
@@ -276,7 +281,7 @@ defmodule IntelliJElixir.Debugger.Server do
               end
 
             # https://github.com/elixir-lang/elixir/blob/8a971fcb44391bd8b16456666f3033b633c6ff77/lib/elixir/src/elixir.erl#L256
-            {erl, _new_env, _new_scope} = quoted_to_erl(quoted, current_vars_env, parsed_scope)
+            erl = quoted_to_erl(quoted, current_vars_env, parsed_scope)
 
             code =
               [:erl_pp.expr(erl), ?.]
@@ -337,10 +342,18 @@ defmodule IntelliJElixir.Debugger.Server do
     :int.meta(meta_pid, :bindings, level)
   end
 
-  defp counter(elixir_variable_tuples) when is_list(elixir_variable_tuples) do
-    Enum.into(elixir_variable_tuples, %{}, fn {elixir_variable_name, counter, _} ->
-      {elixir_variable_name, counter}
-    end)
+  if Version.compare(System.version(), "1.10.0") == :lt do
+    defp counter(elixir_variable_tuples) when is_list(elixir_variable_tuples) do
+      Enum.into(elixir_variable_tuples, %{}, fn {elixir_variable_name, counter, _} ->
+        {elixir_variable_name, counter}
+      end)
+    end
+  else
+    defp counter(elixir_variable_tuples) when is_list(elixir_variable_tuples) do
+      Enum.into(elixir_variable_tuples, %{}, fn {elixir_variable_name, counter, _} ->
+        {elixir_variable_name, counter - 1}
+      end)
+    end
   end
 
   defp elixir_module_name_to_erlang_module_name(":" <> erlang_module_name), do: erlang_module_name
@@ -391,17 +404,27 @@ defmodule IntelliJElixir.Debugger.Server do
   end
 
   # `:elixir.quoted_to_erl/3` became private in Elixir 1.8, so need to inline it here.
-  if Version.compare(System.version(), "1.8.0") == :lt do
-    defp quoted_to_erl(quoted, env, scope) do
-      :elixir.quoted_to_erl(quoted, env, scope)
-    end
-  else
-    defp quoted_to_erl(quoted, env, scope) do
-      {expanded, new_env} = :elixir_expand.expand(quoted, env)
-      {erl, new_scope} = :elixir_erl_pass.translate(expanded, scope)
+  cond do
+    Version.compare(System.version(), "1.8.0") == :lt ->
+      defp quoted_to_erl(quoted, env, scope) do
+        {erl, _, _} = :elixir.quoted_to_erl(quoted, env, scope)
 
-      {erl, new_env, new_scope}
-    end
+        erl
+      end
+    Version.compare(System.version(), "1.13.0") == :lt ->
+      defp quoted_to_erl(quoted, env, scope) do
+        {expanded, _} = :elixir_expand.expand(quoted, env)
+        {erl, _} = :elixir_erl_pass.translate(expanded, scope)
+
+        erl
+      end
+    true ->
+      defp quoted_to_erl(quoted, env, scope) do
+        {expanded, _, _} = :elixir_expand.expand(quoted, :elixir_env.env_to_ex(env), env)
+        {erl, _} = :elixir_erl_pass.translate(expanded, :erl_anno.new(env.line), scope)
+
+        erl
+      end
   end
 
   defp time_interpret(module) when is_atom(module) do
@@ -539,6 +562,33 @@ defmodule IntelliJElixir.Debugger.Server do
       Enum.into(elixir_variable_tuples, %{}, fn {elixir_variable_name, _counter, erlang_variable_name} ->
         {{elixir_variable_name, nil}, {0, erlang_variable_name}}
       end)
+    end
+  end
+
+  if Version.compare(System.version(), "1.7.0") == :lt do
+    defp elixir_env_with_vars(env, parsed_vars) do
+      # https://github.com/elixir-lang/elixir/blob/v1.6.5/lib/elixir/src/elixir.erl#L223
+      %{env | vars: parsed_vars}
+    end
+  else
+    defp elixir_env_with_vars(env, parsed_vars) do
+      :elixir_env.with_vars(env, parsed_vars)
+    end
+  end
+
+  if Version.compare(System.version(), "1.10.0") == :lt do
+    defp parsed_scope(elixir_variable_tuples) do
+      elixir_erl(
+        vars: vars(elixir_variable_tuples),
+        counter: counter(elixir_variable_tuples)
+      )
+    end
+  else
+    defp parsed_scope(elixir_variable_tuples) do
+      elixir_erl(
+        var_names: vars(elixir_variable_tuples),
+        counter: counter(elixir_variable_tuples)
+      )
     end
   end
 end

--- a/resources/debugger/lib/intellij_elixir/debugger/server.ex
+++ b/resources/debugger/lib/intellij_elixir/debugger/server.ex
@@ -7,7 +7,7 @@ defmodule IntelliJElixir.Debugger.Server do
 
   # *.hrl files are not included in all SDK installs, so need to inline definition here
 
-  cond  do
+  cond do
     Version.compare(System.version(), "1.7.0") == :lt ->
       # https://github.com/elixir-lang/elixir/blob/v1.5.0/lib/elixir/src/elixir.hrl#L7-L17
       Record.defrecordp(
@@ -22,6 +22,7 @@ defmodule IntelliJElixir.Debugger.Server do
         counter: %{},
         file: "nofile"
       )
+
     Version.compare(System.version(), "1.9.2") == :lt ->
       # https://github.com/elixir-lang/elixir/blob/v1.7.0/lib/elixir/src/elixir.hrl#L7-L16
       Record.defrecordp(
@@ -35,6 +36,7 @@ defmodule IntelliJElixir.Debugger.Server do
         counter: %{},
         stacktrace: false
       )
+
     true ->
       # https://github.com/elixir-lang/elixir/blob/v1.9.2/lib/elixir/src/elixir.hrl#L8-L18
       Record.defrecordp(

--- a/src/org/elixir_lang/espec/Configuration.kt
+++ b/src/org/elixir_lang/espec/Configuration.kt
@@ -50,7 +50,7 @@ class Configuration(name: String, project: Project) :
 
         // Explicit MIX_ENV so that `intellij_elixir.debug` uses the test code paths
         val envs = mutableMapOf<String, String>()
-        envs.putAll(envs)
+        envs.putAll(this.envs)
         envs.putIfAbsent("MIX_ENV", "test")
         debugged.envs = envs
 

--- a/src/org/elixir_lang/exunit/Configuration.kt
+++ b/src/org/elixir_lang/exunit/Configuration.kt
@@ -50,7 +50,7 @@ class Configuration(name: String, project: Project) :
 
         // Explicit MIX_ENV so that `intellij_elixir.debug` uses the test code paths
         val envs = mutableMapOf<String, String>()
-        envs.putAll(envs)
+        envs.putAll(this.envs)
         envs.putIfAbsent("MIX_ENV", "test")
         debugged.envs = envs
 

